packages/worker/src/token/tokenOffChainData/providers/coingecko/coingeckoTokenOffChainDataProvider.ts

```diff
@@ -49,49 +50,52 @@ export class CoingeckoTokenOffChainDataProvider implements TokenOffChainDataProv
     bridgedTokensToInclude: string[];
   }): Promise<ITokenOffChainData[]> {
     const tokensList = await this.getTokensList();
-    // Include ETH, all zksync L2 tokens and bridged tokens
+    if (!tokensList?.length) {
+      return [];
+    }
+
+    // Include ETH, all chain-specific tokens and bridged tokens
     const supportedTokens = tokensList.filter(
       (token) =>
         token.id === "ethereum" ||
-        token.platforms.zksync ||
-        bridgedTokensToInclude.find((bridgetTokenAddress) => bridgetTokenAddress === token.platforms.ethereum)
+        (token.platforms &&
+          (token.platforms[this.chainId] ||
+            bridgedTokensToInclude.some(
+              (bridgetTokenAddress) =>
+                bridgetTokenAddress.toLowerCase() === (token.platforms.ethereum || "").toLowerCase()
+            )))
     );
 
-    const tokensOffChainData: ITokenOffChainData[] = [];
-    let tokenIdsPerRequest = [];
-    for (let i = 0; i < supportedTokens.length; i++) {
-      tokenIdsPerRequest.push(supportedTokens[i].id);
-      if (tokenIdsPerRequest.length === API_NUMBER_OF_TOKENS_PER_REQUEST || i === supportedTokens.length - 1) {
-        const tokensMarkedData = await this.getTokensMarketData(tokenIdsPerRequest);
-        tokensOffChainData.push(
-          ...tokensMarkedData.map((tokenMarketData) => {
-            const token = supportedTokens.find((t) => t.id === tokenMarketData.id);
-            return {
-              l1Address: token.id === "ethereum" ? utils.ETH_ADDRESS : token.platforms.ethereum,
-              l2Address: token.platforms.zksync,
-              liquidity: tokenMarketData.market_cap,
-              usdPrice: tokenMarketData.current_price,
-              iconURL: tokenMarketData.image,
-            };
-          })
-        );
-        tokenIdsPerRequest = [];
-      }
+    if (!supportedTokens.length) {
+      return [];
     }
-    return tokensOffChainData;
-  }
 
-  private getTokensMarketData(tokenIds: string[]) {
-    return this.makeApiRequestRetryable<ITokenMarketDataProviderResponse[]>({
-      path: "/coins/markets",
-      query: {
-        vs_currency: "usd",
-        ids: tokenIds.join(","),
-        per_page: tokenIds.length.toString(),
-        page: "1",
-        locale: "en",
-      },
+    const tokenIds = supportedTokens.map((token) => token.id);
+    const tokensMarkedData = await this.makeApiRequest<ITokenMarketDataProviderResponse[]>("/coins/markets", {
+      vs_currency: "usd",
+      ids: tokenIds.join(","),
+      per_page: tokenIds.length.toString(),
+      page: "1",
+      locale: "en",
     });
```

Comment:
1. The previous implementation respected the API response's max page size of 250, which is why it requested information about tokens in batches of up to 250 token ids each. This PR removes this logic and sends a single request, which may not be sufficient, especially on Era mainnet.

2. It also used `makeApiRequestRetryable` function, which handles `HTTP 429 Too Many Requests` and implements retries with gradually increasing timeouts. The new functionality no longer uses this approach, which makes the processing less reliable.